modules/javafx.controls/src/test/java/test/javafx/css/PropertySizeTest.java

```diff
@@ -0,0 +1,535 @@
+        testFontSizeOfParents(ROOT_FONT_SIZE + "px", "100px", "80px", "60px", "40px");
+
+        testFontSizeOfParents(ROOT_FONT_SIZE + "em", "0.9em", "0.8em", "0.7em", "0.6em");
+        testFontSizeOfParents(ROOT_FONT_SIZE + "px", "0.9em", "0.8em", "0.7em", "0.6em");
+
+        testFontSizeOfParents(ROOT_FONT_SIZE + "em", "0.9em", "80px", "0.6em", "40px");
+        testFontSizeOfParents(ROOT_FONT_SIZE + "px", "0.9em", "80px", "0.6em", "40px");
+    }
+
+    // This is a specific combination where -fx-font-size of parents is same and test fails.
+    // If we fix this then the test can be moved inside previous test.
+    @Ignore()
+    @Test
+    public void sameRelativeFontSizeOfNestedParentsTest() {
+        testFontSizeOfParents(ROOT_FONT_SIZE + "px", "0.5em", "0.5em",
+                "0.5em", "0.5em");
+    }
+
+    private void testFontSizeOfParents(String rtSize, String p1Size, String p2Size, String p3Size, String p4Size) {
+
+        Property rtFont = new Property(rtSize);
+        Property p1Font = new Property(p1Size);
+        Property p2Font = new Property(p2Size);
+        Property p3Font = new Property(p3Size);
+        Property p4Font = new Property(p4Size);
+
+        String rtStyle = rtSize.equals("") ? "" : "-fx-font-size: " + rtSize;
+        String p1Style = p1Size.equals("") ? "" : "-fx-font-size: " + p1Size;
+        String p2Style = p2Size.equals("") ? "" : "-fx-font-size: " + p2Size;
+        String p3Style = p3Size.equals("") ? "" : "-fx-font-size: " + p3Size;
+        String p4Style = p4Size.equals("") ? "" : "-fx-font-size: " + p4Size;
+
+        root.setStyle(rtStyle);
+        p1.setStyle(p1Style);
+        p2.setStyle(p2Style);
+        p3.setStyle(p3Style);
+        p4.setStyle(p4Style);
+
+        double defFontSize = Font.getDefault().getSize();
+        double rtFontSize = rtFont.getValue(defFontSize);
+        double p1RefeFont = (rtFont.isRelative && p1Font.isRelative) ? defFontSize : rtFontSize;
+        double p1FontSize = p1Font.getValue(p1RefeFont);
+
+        double p2RefeFont = (p1Font.isRelative && p2Font.isRelative) ? rtFontSize : p1FontSize;
+        double p2FontSize = p2Font.getValue(p2RefeFont);
+        //double p2FontSize = p2Font.getValue(rtFontSize);
+
+        double p3RefeFont = (p2Font.isRelative && p3Font.isRelative) ? p1FontSize : p2FontSize;
+        double p3FontSize = p3Font.getValue(p3RefeFont);
+        //double p3FontSize = p3Font.getValue(p1FontSize);
+
+        double p4RefeFont = (p3Font.isRelative && p4Font.isRelative) ? p2FontSize : p3FontSize;
+        double p4FontSize = p4Font.getValue(p4RefeFont);
+        //double p4FontSize = p4Font.getValue(p2FontSize);
+
+        System.err.println("rtFontSize: " + rtFontSize);
+        System.err.println("p1FontSize: " + p1FontSize);
+        System.err.println("p2FontSize: " + p2FontSize);
+        System.err.println("p3FontSize: " + p3FontSize);
+        System.err.println("p4FontSize: " + p4FontSize);
```

Comment:
Are you planning to leave these print statements in? It seems like noise now that you are done debugging the test.

Reply:
They are removed now, I overlooked when cleaning up.